### PR TITLE
Network USM : avoid ebpf maps contentions

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

TLDR all maps with `conn_tuple_t` as key must be edited by the loader to MaxConnectionTracked (65536 by default)

Avoiding hash_lru_map contention (+50% system cpu on user pod) due to map too small compare the numbers of connections
On staging setup : >8000 sockets running on 16 cores, on the packets receive path (socket/classifier)

The main issue is the kernel spinlock an internal LRU list for evicted elements, this list is shared with all cores

* Moved the ebpf maps to the ebpf program that instantiate them as they not shared
* Only instantiate on map once in memory

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

**Review by p-lambert — APPROVED** [review]

File: pkg/network/tracer/connection/kprobe/tracer.go [hunk 1]
```diff
@@ -303,18 +307,15 @@ func (t *kprobeTracer) Stop() {
 
 func (t *kprobeTracer) GetMap(name string) *ebpf.Map {
 	switch name {
+	case string(probes.ConnectionStatesMap):
```
Comment (hmahmood) [inline pkg/network/tracer/connection/kprobe/tracer.go:310]: Do you need the switch statement anymore?

File: pkg/network/ebpf/c/tracer-maps.h [hunk 2]
```diff
@@ -92,4 +93,18 @@ BPF_HASH_MAP(do_sendfile_args, __u64, struct sock *, 1024)
...
+// Some protocol can be classified in a single direction (for example HTTP/2 can be classified only by the first 24 bytes
+// sent on the hand shake), and if we have NAT, then the conn tuple we extract from sk_buff will be different than the
+// one we extract from the sock object, and then we are not able to correctly classify those protocols.
+// To overcome those problems, we save two maps that translates from conn tuple of sk_buff to conn tuple of sock* and vice
+// versa (the vice versa is used for cleanup purposes).
+BPF_HASH_MAP(conn_tuple_to_socket_skb_conn_tuple, conn_tuple_t, conn_tuple_t, 0)
+
+// Maps a connection tuple to its classified protocol. Used to reduce redundant classification procedures on the same
+// connection. Assumption: each connection has a single protocol.
+BPF_HASH_MAP(connection_protocol, conn_tuple_t, protocol_t, 0)
```
Comment (guyarb) [inline pkg/network/ebpf/c/tracer-maps.h:108]: the separation of the maps was intended to make usm as the owner of them and to reduce the requirment of networks team approval for classification changes

File: pkg/network/ebpf/c/tracer-shared-maps.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#ifndef __TRACER_SHARED_MAPS_H
+#define __TRACER_SHARED_MAPS_H
+
+#include "map-defs.h"
+
+// Maps a connection tuple to latest tcp segment we've processed. Helps to detect same packets that travels multiple
+// interfaces or retransmissions.
+BPF_HASH_MAP(connection_states, conn_tuple_t, u32, 0)
```
Comment (guyarb) [inline pkg/network/ebpf/c/tracer-shared-maps.h:8]: the separation of the maps was intended to make usm as the owner of them and to reduce the requirment of networks team approval for classification changes

File: pkg/network/ebpf/c/protocols/protocol-classification-maps.h [hunk 4]
```diff
@@ -5,24 +5,6 @@
...
-// sent on the hand shake), and if we have NAT, then the conn tuple we extract from sk_buff will be different than the
-// one we extract from the sock object, and then we are not able to correctly classify those protocols.
-// To overcome those problems, we save two maps that translates from conn tuple of sk_buff to conn tuple of sock* and vice
-// versa (the vice versa is used for cleanup purposes).
-BPF_HASH_MAP(conn_tuple_to_socket_skb_conn_tuple, conn_tuple_t, conn_tuple_t, 1024)
-
-// Maps a connection tuple to latest tcp segment we've processed. Helps to detect same packets that travels multiple
-// interfaces or retransmissions.
-BPF_HASH_MAP(connection_states, conn_tuple_t, u32, 1024)
-
```
Comment (guyarb) [inline pkg/network/ebpf/c/protocols/protocol-classification-maps.h:25]: the separation of the maps was intended to make usm as the owner of them and to reduce the requirment of networks team approval for classification changes

**Review by guyarb — CHANGES_REQUESTED** [review]

**Review by guyarb — APPROVED** [review]

File: pkg/network/ebpf/c/protocols/protocol-classification-maps.h (hunk 4 above)
Comment (nplanel) [reply pkg/network/ebpf/c/protocols/protocol-classification-maps.h:25]: as discuss offline, this as been move to protocol-classification-tracer-maps.h as it's used by the tracer

File: pkg/network/tracer/connection/kprobe/tracer.go (hunk 1 above)
Comment (nplanel) [reply pkg/network/tracer/connection/kprobe/tracer.go]: the switch is used as a filter to all the real ebpf.GetMap()
